src/server/auth/authMiddleware/index.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import { AUTH_WHITELIST, GRAPHQL_ENDPOINT } from '@constants/api'
+import TokenModel from '@models/Token'
+import type { User } from '@models/User'
+import database from '@server/database'
+
+const authMiddleware: express$Middleware = async (
+  request: express$Request,
+  response: express$Response,
+  next: express$NextFunction
+) => {
+  if (AUTH_WHITELIST.indexOf(request.path) !== -1) {
+    return next()
+  }
+
+  if (
+    typeof request.headers.authorization === 'undefined' ||
+    request.headers.authorization === null
+  ) {
+    if (request.path === GRAPHQL_ENDPOINT) {
+      request.user = null
+
+      return next()
+    } else {
```

Comment:
You don’t need the else if you return in the `if` above. Nesting makes things more complicated to read.